lib/polyfill/patchedmediakeys_20140218.js

```diff
@@ -0,0 +1,832 @@
+        var cap = cfg.videoCapabilities[j];
+        if (cap.contentType) {
+          ranAnyTests = true;
+          var contentType = cap.contentType.split(';')[0];
+          if (MSMediaKeys.isTypeSupported(this.keySystem, contentType)) {
+            newCfg.videoCapabilities.push(cap);
+            success = true;
+          }
+        }
+      }
+    }
+
+    if (!ranAnyTests) {
+      // If no specific types were requested, we check all common types to find
+      // out if the key system is present at all.
+      success = MSMediaKeys.isTypeSupported(this.keySystem, 'video/mp4');
+    }
+    if (cfg.persistentState == 'required') {
+      if (allowPersistentState) {
+        newCfg.persistentState = 'required';
+        newCfg.sessionTypes = ['persistent-license'];
+      } else {
+        success = false;
+      }
+    }
+
+    if (success) {
+      this.configuration_ = newCfg;
+      return;
+    }
+  }  // for each cfg in supportedConfigurations
+
+  // As per the spec, this should be a DOMException, but
+  // there is not a public constructor for this
+  var unsupportedKeySystemError = new Error('Unsupported keySystem');
+  unsupportedKeySystemError.name = 'NotSupportedError';
+  unsupportedKeySystemError.code = DOMException.NOT_SUPPORTED_ERR;
+  throw unsupportedKeySystemError;
+};
+
+
+/** @override */
+shaka.polyfill.PatchedMediaKeys.v20140218.MediaKeySystemAccess.prototype.
+    createMediaKeys = function() {
+  shaka.log.debug('v20140218.MediaKeySystemAccess.createMediaKeys');
+
+  // Alias
+  var v20140218 = shaka.polyfill.PatchedMediaKeys.v20140218;
+
+  var mediaKeys = new v20140218.MediaKeys(this.keySystem);
+  return Promise.resolve(/** @type {!MediaKeys} */ (mediaKeys));
+};
+
+
+/** @override */
+shaka.polyfill.PatchedMediaKeys.v20140218.MediaKeySystemAccess.prototype.
+    getConfiguration = function() {
+  shaka.log.debug('v20140218.MediaKeySystemAccess.getConfiguration');
+  return this.configuration_;
+};
+
+
+/**
+ * An implementation of HTMLMediaElement.prototype.setMediaKeys.
+ * Attach a MediaKeys object to the media element.
+ *
+ * @this {!HTMLMediaElement}
+ * @param {MediaKeys} mediaKeys
+ * @return {!Promise}
+ */
+shaka.polyfill.PatchedMediaKeys.v20140218.setMediaKeys = function(mediaKeys) {
+  shaka.log.debug('v20140218.setMediaKeys');
+
+  // If mediaKeys is null, ensure mediaKeys on the videoElement is null
+  if (!mediaKeys) {
+    delete this['mediaKeys'];
+    this['mediaKeys'] = null;
+
+    return Promise.reject('MediaKeys not supplied');
+  }
+
+  // If mediaKeys equals videoElement's mediaKey instance, do nothing
+  if (this.mediaKeys == mediaKeys) {
+    return Promise.resolve();
+  }
+
+  // Alias
+  var v20140218 = shaka.polyfill.PatchedMediaKeys.v20140218;
+
+  delete this['mediaKeys'];  // in case there is an existing getter
+  this['mediaKeys'] = mediaKeys;  // work around read-only declaration
+
+  return mediaKeys.setMedia(this);
```

Comment:
Since you aren't calling setMedia(null) on the old value of mediaKeys, there seems to be no way for this polyfill to remove listeners created by setMedia.  This could lead to a memory leak by holding references to the media element.